[IsabelChong] iP

IsabelChong opened the pull request:
# 🍒 Berry the Bunny Bot ૮ ˶ᵔ ᵕ ᵔ˶ ა

> Not having enough sleep ~~because of -~~ makes us  forgetful... 

## _🐰 About you..._

- always busy with lots of tasks to do
- write down your tasks somewhere but lose them :<
- **forgetful?**

### 🍒 Berry is here to help you keep track of your tasks 🍒 
Get it in just a few steps!

1. download it from [here](https://github.com/IsabelChong/ip/releases/tag/v0.2)
2. add your tasks
3. let it manage your tasks for you! ૮₍ ˶ᵔ ᵕ ᵔ˶ ₎ა

And it is **FREE!!**

### 🍒Features:🍒

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

___

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

IsabelChong committed:
Duke.java: Requirements of Level-1 added in.
- added in chatbot personality "Berry"
- added in the function of repeating back the command given to it.
- added in the function of handling "bye" command and other commands.

IsabelChong committed:
Duke.java: modified repeat function to become addText.

IsabelChong committed:
Duke.java: added functionality to store and list tasks, completing Level-2 requirements.

IsabelChong committed:
Duke.java: Requirements of Level-3 added in.
- added in Task and TaskBook class.

IsabelChong committed:
Duke.java: fixed bugs and added in javadocs with comments.

IsabelChong committed:
Duke.java: changed access modifiers and function names.

IsabelChong committed:
Duke.java: fixed some naming conventions and cleaned code.

IsabelChong committed:
Duke.java: added in support for child classes for Tasks (Todo, Deadline, Event).

IsabelChong committed:
Duke.java: bug fixes, can handle common Task input.

IsabelChong committed:
Duke.java: minor grammar fixes.

IsabelChong committed:
automated testing input.txt and EXPECTED.txt updated

IsabelChong committed:
Duke.java: added in skeleton for DukeException class.

IsabelChong committed:
Duke.java: added exception handling for out of range inputs, missing inputs, and unreadable commands.

IsabelChong committed:
Duke.java: bug fixes and and added deleteTask function.

IsabelChong committed:
Duke.java: updated javadocs

IsabelChong committed:
Duke.java: added enum TaskType to indicate task types.

IsabelChong committed:
Updated input.txt and EXPECTED.txt for automated testing.

IsabelChong committed:
Separate Duke.java into Duke.java, Deadline.java, Task.java, TaskType.java, Todo.java, Event.java

There were several classes in Duke.java which made it hard to read and debug.

Separating them into their own java class files improves code readability.

Let's create new java files for their respective new classes.

IsabelChong committed:
Separate DukeException.java into EmptyDescriptionException.java, IndexOutOfRangeException.java, MissingClauseException.java and UnknownCommandException.java

There were several types of exceptions to handle which is more clearly depicted by separate classes which inherit DukeException.

Separating them into their own java class exception files improves code readability.

Let's create new java exception files for their respective new classes.

IsabelChong committed:
Update expected output

IsabelChong committed:
Add Storage.java

IsabelChong committed:
Add constructor to handle done status when reading from existing file

IsabelChong committed:
Add interpreter methods for retrieving from file or saving to file.

IsabelChong committed:
Update main to handle file storage and retrieval

IsabelChong committed:
Remove static counter and add getter for listOfTasks

IsabelChong committed:
Add new exception to handle incorrect date format

IsabelChong committed:
Change date variables from String to LocalDate.

IsabelChong committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Change date variables from String to LocalDate.
  Add new exception to handle incorrect date format

IsabelChong committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

IsabelChong committed:
Add Parser class and Command classes

IsabelChong committed:
Extract code into Ui, Storage and TaskList class

IsabelChong committed:
Rename Duke to Berry

IsabelChong committed:
Organise classes into packages

IsabelChong committed:
Update method names to comply with coding standard

IsabelChong committed:
Add JUnit tests

IsabelChong committed:
Update exceptions for invalid file names

IsabelChong committed:
Add JavaDoc comments

IsabelChong committed:
Tweak the JUnit code to comply with a coding standard

IsabelChong committed:
Add manifest version

IsabelChong committed:
Tweak the command codes to comply with a coding standard

IsabelChong committed:
Tweak code to comply with a coding standard

IsabelChong committed:
Add Javadoc comments to exception classes

IsabelChong committed:
Removed wildcard imports

IsabelChong committed:
Add Javadoc for class members

IsabelChong committed:
Add find command to search for keyword in tasks

IsabelChong committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Removed wildcard imports
  Tweak code to comply with a coding standard
  Tweak the command codes to comply with a coding standard
  Tweak the JUnit code to comply with a coding standard

IsabelChong committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add Javadoc for class members
  Add Javadoc comments to exception classes
  Add JavaDoc comments

# Conflicts:
#	src/main/java/berry/command/AddTaskCommand.java
#	src/main/java/berry/command/Command.java
#	src/main/java/berry/command/DeleteCommand.java
#	src/main/java/berry/command/ExitCommand.java
#	src/main/java/berry/command/ListCommand.java
#	src/main/java/berry/command/MarkCommand.java
#	src/main/java/berry/command/UnmarkCommand.java
#	src/main/java/berry/storage/Storage.java
#	src/main/java/berry/task/Deadline.java
#	src/main/java/berry/task/Event.java
#	src/main/java/berry/task/Task.java
#	src/main/java/berry/task/TaskList.java

IsabelChong committed:
Merge branch 'branch-Level-9'

IsabelChong committed:
Improve naming convention

IsabelChong committed:
Update tasks in storage for Delete, Mark and Unmark commands.

IsabelChong committed:
Add checkstyle configurations

IsabelChong committed:
Add in basic JavaFX with default Duke GUI

Cheamybunny commented on src/main/java/berry/parser/Parser.java:
```diff
@@ -0,0 +1,119 @@
+     * @param input is the full user input string
+     * @throws BerryException if the given string has missing or incomplete arguments
+     */
+    private static void validate(CommandType commandType, String input) throws BerryException {
```
Good job identifying all the potential errors that may occur

Cheamybunny reviewed (commented):
Overall very nicely done and clean code

IsabelChong committed:
Create GUI for Berry

IsabelChong committed:
Updated gradle files

IsabelChong committed:
Restructure code to use GUI

IsabelChong committed:
Update FindCommand to include varargs to find variable keywords

IsabelChong committed:
Merge branch 'branch-A-Varargs'

* branch-A-Varargs:
  Update FindCommand to include varargs to find variable keywords

IsabelChong committed:
Merge branch 'branch-Level-10'

IsabelChong committed:
Remove unnecessary print lines

Johnchiahk commented on src/main/java/berry/command/Command.java:
```diff
@@ -0,0 +1,32 @@
+ */
+public abstract class Command {
+
+    private boolean isExit;
```
Nice choice usage of a boolean variable!

Johnchiahk commented on src/main/java/berry/storage/Storage.java:
```diff
@@ -0,0 +1,111 @@
+import berry.task.Task;
+import berry.task.TaskList;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
```
Great that you didn't do import java.io.* even though there were more than 3 classes you had to import :D

Johnchiahk commented on src/main/java/berry/task/TaskList.java:
```diff
@@ -0,0 +1,104 @@
+
+/**
+ * Contains the task list.
+ */
+public class TaskList {
+
+    private static ArrayList<Task> listOfTasks;
```
The variable name you chose was great!

Johnchiahk commented on src/main/java/berry/command/FindCommand.java:
```diff
@@ -0,0 +1,22 @@
+import berry.exception.BerryException;
+
+public class FindCommand extends Command {
+    private static String[] keyword;
```
Perhaps listOfKeywords may be better?

IsabelChong replied:
Oh right, thanks for this! I will take extra care to rename my variable names when I change their implementations or type ><

Johnchiahk reviewed (commented):
Overall, I found your code very easy to read and OOP principles were implemented well.

liujiayue314 commented on src/main/java/berry/task/TaskList.java:
```diff
@@ -0,0 +1,104 @@
+     *
+     * @param index is the number identifying the task given by the user as seen from the task list
+     */
+    public String markDone(int index) {
```
maybe a better name will be markTaskDone

liujiayue314 commented on src/main/java/berry/task/TaskList.java:
```diff
@@ -0,0 +1,104 @@
+     *
+     * @param index is the number identifying the task given by the user as seen from the task list
+     */
+    public String markNotDone(int index) {
```
similarly maybe can change to markTaskNotDone

liujiayue314 commented on src/main/java/berry/task/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+ */
+public class Deadline extends Task {
+
+    /** The date to do the task by */
```
maybe format should change to
/**
...
*/

or just //....

liujiayue314 commented on src/main/java/berry/task/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+public class Deadline extends Task {
+
+    /** The date to do the task by */
+    protected LocalDate by;
```
any reason why u use protected instead of private, since deadline doesnt have a subclass?

IsabelChong replied:
I think I was just not paying extra attention to the access mods. Thanks for pointing it out! I will make sure to double-check them <:

liujiayue314 commented on src/main/java/berry/parser/Parser.java:
```diff
@@ -0,0 +1,131 @@
+     * @param input is the full user input string
+     * @throws BerryException if the given string has missing or incomplete arguments
+     */
+    private static void validate(CommandType commandType, String input) throws BerryException {
```
maybe a better name will be validateCommand

liujiayue314 reviewed (commented):
Overall u have a clean code, good job!

IsabelChong committed:
Change access modifiers and variable names

The variables in a class with no intended inheritance should be set to private to prevent unintended exposure of the subclass. There are also instances of inappropriate variable names to represent a plural object.

Improving the access modifier from protected to public, and representing the variables in their correct grammar form improves code quality.

This improves readability so that the code can be easily maintained and referred.

Let's,
* change the access modifiers of variables in classes with no child classes to private
* update variable names to better suit its object type - singular or plural, especially if there are any changes made to its implementations

IsabelChong committed:
Add assert feature into code

There are several assumptions that have been made for the functionality of several methods. There should be a way to document the assumptions at various points in the code.

Assertions document the important assumptions made for readability.

Let's,
* add assert statements for relevant assumptions made for a method functionality

IsabelChong committed:
Update variables names to comply with code quality

IsabelChong committed:
Merge pull request #1 from IsabelChong/branch-A-Assertions

Add Java assert statements

IsabelChong committed:
Merge branch 'master' into branch-A-CodeQuality

IsabelChong committed:
Merge pull request #2 from IsabelChong/branch-A-CodeQuality

Update variables names to comply with code quality

IsabelChong committed:
Update GUI

IsabelChong committed:
Add assert to execute commands

IsabelChong committed:
Restructure code to meet code quality

IsabelChong committed:
Restructure code to fit SLAP

IsabelChong committed:
Add tagging for tasks

IsabelChong committed:
Add and set up Continuous Integration (CI)

IsabelChong committed:
Update User Interface

The user interface is messy and hard to read with striking borders and dull background.

Using an easy-to-the-eye colour and rounding up the edges of images and text boxes makes it easier for users to read.

Let's,
* update the UI to better suit the user's needs of the UI being easy to look at

IsabelChong committed:
Refactor codes to appropriate directories

IsabelChong committed:
Update GUI

IsabelChong committed:
Update GUI

IsabelChong committed:
Add files via upload

IsabelChong committed:
Update GUI

IsabelChong committed:
Merge branch 'master' of https://github.com/IsabelChong/ip

* 'master' of https://github.com/IsabelChong/ip:
  Add files via upload

IsabelChong committed:
Update README.md

IsabelChong committed:
Fixes to parsing of find commands

IsabelChong committed:
Add UI Image

IsabelChong committed:
Update README.md

IsabelChong committed:
Update Ui image

IsabelChong committed:
Update UI image

IsabelChong committed:
Update FindCommand.java output structure

IsabelChong committed:
Add in HelpCommand

IsabelChong committed:
Update README.md

IsabelChong committed:
Update README.md

IsabelChong committed:
Update README.md

IsabelChong committed:
Restructure code to fit Code Quality with checkstyle

IsabelChong committed:
Merge branch 'master' of https://github.com/IsabelChong/ip

* 'master' of https://github.com/IsabelChong/ip:
  Update README.md
  Update README.md

IsabelChong committed:
Restructure code to fit Code Quality standards

IsabelChong committed:
Rearrange import statements

IsabelChong committed:
Update README.md

IsabelChong committed:
Update DeadlineTest.java

IsabelChong committed:
Update README.md

IsabelChong committed:
Update README.md

IsabelChong committed:
Update test files

IsabelChong committed:
Update README.md

IsabelChong committed:
Update README.md

IsabelChong committed:
Update README.md

IsabelChong committed:
Merge branch 'master' of https://github.com/IsabelChong/ip

* 'master' of https://github.com/IsabelChong/ip:
  Update README.md
  Update README.md

IsabelChong committed:
Update docs/README.md

IsabelChong committed:
Merge branch 'branch-UpdateUG-2'

* branch-UpdateUG-2:
  Update docs/README.md

IsabelChong committed:
Update UserGuide.md

IsabelChong committed:
Update README.md

IsabelChong committed:
Update README.md

IsabelChong committed:
Update README.md
